services/spar/src/Spar/Scim/User.hs

```diff
@@ -762,7 +763,7 @@ scimFindUserByHandle mIdpConfig stiTeam hndl = do
 -- successful authentication with their SAML credentials.
 scimFindUserByEmail :: Maybe IdP -> TeamId -> Text -> MaybeT (Scim.ScimHandler Spar) (Scim.StoredUser ST.SparTag)
 scimFindUserByEmail mIdpConfig stiTeam email = do
-  veid <- mkValidExternalId mIdpConfig (pure email)
+  veid <- MaybeT (either (const Nothing) Just <$> runExceptT (mkValidExternalId mIdpConfig (pure email)))
```

Comment:
It would also be good to add a comment here (or in the test) why not sending an email is accepted here, linking to the relevant documentation.
Our API should not be microsoft compatible and break with other providers (or have less good guarantees) without that explicitly documented in the code.

Reply:
I agree, there has to be a comment here.  I would write:

```suggestion
  -- Azure has been observed to search for externalIds that are not emails, even if the
  -- mapping is set up like it should be.  This is a problem: if there is no SAML IdP, 'mkValidExternalId'
  -- only supports external IDs that are emails.  This is a missing feature / bug in spar tracked in
  -- https://wearezeta.atlassian.net/browse/SQSERVICES-157; once it is fixed, we should go back to
  -- throwing errors returned by 'mkValidExternalId' here, but *not* throw an error if the externalId is
  -- a UUID, or any other text that is valid according to SCIM.
  veid <- MaybeT (either (const Nothing) Just <$> runExceptT (mkValidExternalId mIdpConfig (pure email)))
```